src/refdb_fs.c

```diff
@@ -1099,22 +1099,27 @@ static int packed_write(refdb_fs_backend *backend)
 static int packed_delete(refdb_fs_backend *backend, const char *ref_name)
 {
 	size_t pack_pos;
-	int error;
+	int error, found = 0;
 
 	if ((error = packed_reload(backend)) < 0)
 		goto cleanup;
 
-	/* If a packed reference exists, remove it from the packfile and repack */
 	if ((error = git_sortedcache_wlock(backend->refcache)) < 0)
 		goto cleanup;
 
-	if (!(error = git_sortedcache_lookup_index(
-			&pack_pos, backend->refcache, ref_name)))
+	/* If a packed reference exists, remove it from the packfile and repack if necessary */
+	error = git_sortedcache_lookup_index(&pack_pos, backend->refcache, ref_name);
+	if (error == 0) {
 		error = git_sortedcache_remove(backend->refcache, pack_pos);
+		found = 1;
+	}
+	if (error == GIT_ENOTFOUND)
+		error = 0;
 
 	git_sortedcache_wunlock(backend->refcache);
 
-	error = packed_write(backend);
+	if (found)
+		error = packed_write(backend);
```

Comment:
I wonder if it's worth making sure that `GIT_ENOTFOUND` could not be propagated back from either `git_sortedcache_wlock` or `packed_write`.  A quick glance suggests that those will never return those, and it's hard to imagine a world where they would be refactored to a point where `GIT_ENOTFOUND` would be meaningful, but it feels Not Impossible.  So we might want to be strict about converting those to a `-1` if we want to treat them as a true failure here.  🤔 

OTOH, it will probably never matter.

Reply:
That… went over my head. What's the concern ? That a `GIT_ENOTFOUND` from `wlock` or `write` could be misinterpreted as meaning the ref doesn't exist, while the issue could be more severe ?